kerberos-integration/keytab-creator.ps1

```diff
@@ -1,97 +1,137 @@
-			$ErrorActionPreference = $ExistingEAP
-			$usercheck_success = $true
-			Write-Output 'Successfully generated new AD user '$ad_username
-		}
-		catch 
-		{
-			$ErrorActionPreference = $ExistingEAP
-    			Write-Warning "Failed to create user: $($error[0])"
-		}
-	}	
+    Write-Output 'Generating new AD user:'
+    while (!$usercheck_success)
+    {
+        $ExistingEAP = $ErrorActionPreference
+        try{
+            $ad_username = Read-Host -Prompt 'AD username / sAMAccountName (e.g., aduser)'
+            $ad_user_psw = Read-Host -Prompt 'AD user password' -AsSecureString
+            $credential = New-Object System.Net.NetworkCredential($ad_username, $ad_user_psw)
+            
+            $ErrorActionPreference = "Stop"
+            New-ADUser -Name $ad_username -PasswordNeverExpires $true -AccountPassword $ad_user_psw -KerberosEncryptionType 16 -Enabled $true -ErrorAction Stop
+            $ErrorActionPreference = $ExistingEAP
+            $usercheck_success = $true
+            Write-Output "Successfully generated new AD user $ad_username"
+        }
+        catch 
+        {
+            $ErrorActionPreference = $ExistingEAP
+            Write-Warning "Failed to create user: $($_.Exception.Message)"
+        }
+    }    
 }
 else
 {
-	while (!$usercheck_success)
-	{
-
-		try{
-			$ad_username = Read-Host -Prompt 'AD username / sAMAccountName (e.g., aduser)'
-			$ad_user_psw = Read-Host -AsSecureString 'AD user password'
-			$credential = New-Object System.Net.NetworkCredential($ad_username, $ad_user_psw)
-			Write-Output 'Validating AD user:'
-			Get-ADUser -Identity $ad_username
-			$usercheck_success = $true
-			Write-Output 'Successfully validated AD user '$ad_username
-		}
-		catch 
-		{
-    			Write-Warning "Failed to validate user: $($error[0])"
-		}
-	}	
+    while (!$usercheck_success)
+    {
+        try
+        {
+            $ad_username = Read-Host -Prompt 'AD username / sAMAccountName (e.g., aduser)'
+            $ad_user_psw = Read-Host -Prompt 'AD user password' -AsSecureString
+            $credential = New-Object System.Net.NetworkCredential($ad_username, $ad_user_psw)
+            Write-Output 'Validating AD user:'
+            $user = Get-ADUser -Identity $ad_username -Properties KerberosEncryptionType
+            # Check if the user has AES256-SHA1 encryption enabled
+            $hasAES256 = $user.KerberosEncryptionType -contains "AES256"
+            # Display the appropriate message
+            if ($hasAES256) {
+                Write-Host "User $ad_username has AES256 encryption enabled." -ForegroundColor Green
+            } 
+            else {
+                Write-Warning "Enabled AES256 encryption support for user account $ad_username "
+                Set-ADUser -Identity $ad_username -KerberosEncryptionType AES256
+            }
+            $usercheck_success = $true
```

Comment:
What about the SPN check? Should the service account not have the servicePrincipalName of the cluster registered?